src/router/outputqueued/Router.cc

```diff
@@ -269,20 +269,22 @@ void Router::initialize() {
 
     for (u32 vc = 0; vc < numVcs_; vc++) {
       u32 vcIdx = vcIndex(port, vc);
-      // tell the congestion sensor module of the number of credits in IQ
-      if ((congestionMode_ == Router::CongestionMode::kDownstream) ||
-          (congestionMode_ == Router::CongestionMode::kOutputAndDownstream)) {
+
+      // tell the congestion sensor module of the number of credits
+      if (congestionMode_ == Router::CongestionMode::kDownstream) {
         congestionSensor_->initCredits(vcIdx, credits);
+      } else if (congestionMode_ == Router::CongestionMode::kOutput) {
+        congestionSensor_->initCredits(vcIdx, outputQueueDepth_);
+      } else if (congestionMode_ ==
+                 Router::CongestionMode::kOutputAndDownstream) {
+        congestionSensor_->initCredits(
+            vcIdx,
+            (credits == U32_MAX || outputQueueDepth_ == U32_MAX) ?
+            U32_MAX : credits + outputQueueDepth_);
```

Comment:
this is the second time I've seen this, perhaps a saturating add function would be useful

Reply:
I think this makes it clear that U32_MAX is a special value.